### PR TITLE
Use access token instead for Bangumi tracker

### PR DESCRIPTION
**Pull request by KayZhang52**
Changed the bangumi tracker's authentication to a dialog. User will have to go to the new token site, generate the token and add it here with the expiry date. Does solve the 7 day problem though.
  

screenshot

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -20,7 +20,7 @@ android {
     namespace = "eu.kanade.tachiyomi"
 
     defaultConfig {
-        applicationId = "eu.kanade.tachiyomi"
+        applicationId = "eu.kanade.tachiyomi_bangumi_tracker_expiry"
```
Comment (alessandrojean) [inline app/build.gradle.kts:23]: This shouldn't be commited.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
         android:enableOnBackInvokedCallback="true"
         android:hardwareAccelerated="true"
         android:icon="@mipmap/ic_launcher"
-        android:label="@string/app_name"
+        android:label="Tachiyomi_track_bangumi_fix"
```
Comment (alessandrojean) [inline app/src/main/AndroidManifest.xml:38]: Neither this.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

**Comment by mutsumi63** [comment]
> Changed the bangumi tracker's authentication to a dialog. User will have to go to the new token site, generate the token and add it here with the expiry date. Does solve the 7 day problem though.
> 
> screenshot

In the dialog box, there may be a need for some explanations and URL link.

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (KayZhang52) [reply app/src/main/AndroidManifest.xml]: > Neither this.

Removed. My bad.

**Comment by mutsumi63** [comment]
image

image

Click the tracker again will reset the local records to empty/0.

Click rating button causes a crash.

File: app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsTrackingScreen.kt [hunk 3]
```diff
@@ -281,6 +290,102 @@ object SettingsTrackingScreen : SearchableSettings {
...
+        var processing by remember { mutableStateOf(false) }
+        var inputError by remember { mutableStateOf(false) }
+
+
+        AlertDialog(
+            onDismissRequest = onDismissRequest,
+            title = {
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    Text(
+                        text = "Enter access token for Bangumi",
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsTrackingScreen.kt:318]: Any text in the UI should be added to `strings.xml` instead.

File: app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsTrackingScreen.kt [hunk 4]
```diff
@@ -281,6 +290,102 @@ object SettingsTrackingScreen : SearchableSettings {
...
+                        label = { Text(text = "access token") },
+                        visualTransformation = VisualTransformation.None,
+                        keyboardOptions = KeyboardOptions(
+                            keyboardType = KeyboardType.Text,
+                            imeAction = ImeAction.Done,
+                        ),
+                        singleLine = true,
+                        isError = inputError && !processing,
+                    )
+                    OutlinedTextField(
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsTrackingScreen.kt:344]: Why does the user need to specify this?

File: gradle/kotlinx.versions.toml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [versions]
 kotlin_version = "1.9.10"
 serialization_version = "1.6.0"
-xml_serialization_version = "0.86.2"
+xml_serialization_version = "0.86.1"
```
Comment (arkon) [inline gradle/kotlinx.versions.toml:4]: These dependency downgrades should be removed.

**Review by arkon — CHANGES_REQUESTED** [review]
Seems like @mutsumi63 pointed out some functional issues too?

**Comment by mutsumi63** [comment]
> Seems like @mutsumi63 pointed out some functional issues too?

I added some details.

**Comment by arkon** [comment]
Assuming this is abandoned.

**Pull request closed without merging**